### PR TITLE
chore(ci): frontend openshift deployment

### PR DESCRIPTION
**Pull request by DerekRoberts**
---

Thanks for the PR!

Deployments, as required, will be available below:
- [Frontend](https://nr-sustainment-capstone-2024-119-frontend.apps.silver.devops.gov.bc.ca)
- [Backend](https://nr-sustainment-capstone-2024-119-frontend.apps.silver.devops.gov.bc.ca/api)

Please create PRs in draft mode.  Mark as ready to enable:
- [Analysis Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/analysis.yml)

After merge, new images are deployed in:
- [Merge Workflow](https://github.com/bcgov/nr-sustainment-capstone-2024/actions/workflows/merge.yml)

### CODE REVIEW

**Comment by kcaparas1630** [comment]
You're rocking Derek 🚀

File: backend/Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Base image for the container
-FROM node:20.4.0
+FROM node:20.14.0
```
Comment (GDamaso) [inline backend/Dockerfile:2]: Did it break something  with 20.4? Would like me to investigate some other smaller images to try and save some resources?

File: frontend/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+FROM node:20.14.0
+
+ENV npm_config_cache=/app/.npm
+ENV APP_USER=1011540000
 
 # Copy files, install dependencies and build
+USER ${APP_USER}
 WORKDIR /app/
-COPY . ./
+COPY --chown=${APP_USER} . ./
```
Comment (GDamaso) [inline frontend/Dockerfile:10]: Good to know about the `--chown` flag for the `COPY` here :smile:

**Review by GDamaso — APPROVED** [review]
Thanks for investing your time in helping us out. I'm learning a bit by watching these changes. I'm deffinitly comming back to the workflows and breaking stuff when we get things going, just so I'm sure I know what they're doing. :smile: 
Let me know if there's anything I can help with :+1:

File: backend/Dockerfile (hunk 1 above)
Comment (DerekRoberts) [reply backend/Dockerfile:2]: Its just keeping up to date. If nothing breaks, always step up.

File: frontend/Dockerfile (hunk 2 above)
Comment (DerekRoberts) [reply frontend/Dockerfile:10]: It's great! Thanks for the comments. :D

**Comment by DerekRoberts** [comment]
@GDamaso Sorry, jumped the gun! Going back to draft. You're not require to know DevOps, but I'm happy to share. Just as soon as I get the damn thing working consistently!!

File: backend/Dockerfile (hunk 1 above)
Comment (DerekRoberts) [reply backend/Dockerfile:2]: BTW smaller images are a great idea!  We have "ways" of chopping down too, but that falls under optimization.

**Pull request merged**